crates/starknet-server/src/api/json_rpc/endpoints.rs

```diff
@@ -109,9 +112,59 @@ impl JsonRpcHandler {
     /// starknet_getTransactionByHash
     pub(crate) async fn get_transaction_by_hash(
         &self,
-        _transaction_hash: TransactionHashHex,
+        transaction_hash: TransactionHashHex,
     ) -> RpcResult<TransactionWithType> {
-        Err(error::ApiError::TransactionNotFound)
+        let starknet = self.api.starknet.read().await;
+        let transaction_to_map = starknet.transactions.get(&transaction_hash.0).unwrap();
+        let transaction_type;
+        let transaction_data: Transaction = match transaction_to_map.inner.clone() {
+            starknet_core::transactions::Transaction::Declare(declare_v1) => {
+                transaction_type = TransactionType::Declare;
+
+                // Is case of a reverted transaction we need to set class_hash.
```

Comment:
Typo: Is -> In

If we need to remember this, it's not good. We can create an issue for this. Or you can create an issue for 0.4.0 RPC update and mention this there.

Reply:
I just removed this comment and used `unwrap_or_default()` so now we will have a compilation error when we will change the type during adoption.